File: mobile/modules/caddieinsight-storage/ios/CaddieInsightStorageModule.swift
import ExpoModulesCore
import Foundation

/**
 * Stub native module. Task 1 scaffold only — protectAndVerify / ensureProtectedRoots
 * throw until full Application Support + FileProtectionType.complete implementation lands.
 */
public class CaddieInsightStorageModule: Module {
  public func definition() -> ModuleDefinition {
    Name("CaddieInsightStorage")

    AsyncFunction("ensureProtectedRoots") { () -> [String: String] in
      throw Exception(
        name: "CaddieInsightStorageUnavailable",
        description: "CaddieInsightStorage ensureProtectedRoots is not implemented in this scaffold stub."
      )
    }

    AsyncFunction("protectAndVerify") { (_ uri: String) in
      throw Exception(
        name: "CaddieInsightStorageUnavailable",
        description: "CaddieInsightStorage protectAndVerify is not implemented in this scaffold stub."
      )
    }
  }
}

File: mobile/modules/caddieinsight-storage/android/src/main/java/expo/modules/caddieinsightstorage/CaddieInsightStorageModule.kt
package expo.modules.caddieinsightstorage

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.exception.CodedException

/**
 * Stub native module. Task 1 scaffold — fail closed until noBackupFilesDir roots land.
 */
class CaddieInsightStorageModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("CaddieInsightStorage")

    AsyncFunction("ensureProtectedRoots") {
      throw CodedException(
        "CaddieInsightStorageUnavailable",
        "CaddieInsightStorage ensureProtectedRoots is not implemented in this scaffold stub.",
        null
      )
    }

    AsyncFunction("protectAndVerify") { _: String ->
      throw CodedException(
        "CaddieInsightStorageUnavailable",
        "CaddieInsightStorage protectAndVerify is not implemented in this scaffold stub.",
        null
      )
    }
  }
}
